cli/concept.md

```diff
@@ -0,0 +1,157 @@
+- amplify-category-notifications
+- amplify-category-storage
+- amplify-codegen
+- amplify-frontend-javascript
+- amplify-frontend-android
+- amplify-frontend-ios
+- amplify-provider-awscloudformation
+
+## Third party plugin setup
+It's easy to add a third party plugin to the Amplify cli. 
+1. The plugin author names the plugin package according to the naming convention outlined above, and publishes it to the npm registry.
+2. The plugin user executes `npm install -g <plugin>` and installs the plugin to the global node_modules directory.
+
+The plugin is then picked up by the cli core and used the same as the official plugins. 
+
+***
+# Commands
+### Official cli commands
+- `amplify categoires`
+- `amplify configure`
+- `amplify console`
+- `amplify delete`
+- `amplify help`
+- `amplify init`
+- `amplify publish`
+- `amplify push`
+- `amplify pull` (to be implemented)
+- `amplify run`
+- `amplify status`
+
+### Most plugins also have these commands
+- `amplify <plugin> configure`
+- `amplify <category> add`
+- `amplify <category> remove`
+- `amplify <category> push`
+- `amplify <category> pull` (to be implemented)
+
+***
+# Typical cli workflow
+The following command should be executed inside the user project's root directory: 
+1. `amplify init`
+2. `amplify <category> add/remove`
+3. `amplify push`
+
+## The init process
+`$ amplify init` <br/>
+The `init` command must be executed at the root directory of a project to initialize the project for the amplify-cli to work with. 
+The `init` command goes through these steps to setup things: 
+- Analyzes the project and confirms with the user to pick the right frontend plugin to handle the project.
+- Carries out the initialization logic of the selected frontend plugin.
+- If there are multiple provider plugins, prompts the user to selected the provider plugins that will provide accesses to backend cloud resources. 
+- Carries out, in sequence, the initialization logic of the selected provider plugin(s). 
+- Insert amplify folder structure into the project's root directory, with the initial project configuration information written in it. 
+- Generate the project metadata file, amplify-meta.json, with the outputs of the above-selected frontend plugin and provider plugin(s). The amplify-meta.json file is inside the amplify folder structure.
+- Generate the .amplifyrc file, it is written to the root directory or the project, outside of the amplify folder structure. 
+
+## Configuration
+### amplify configure
+This command will lead the user to setup a new aws iam user, then save the credentials locally in a named profile, which can then be used by a project for aws access. The user specifies if and what profile is used for a project in the init process, and can later change it using the `amplify configure project` command.
+### amplify configure project
+This command allows the user to change the project configuration set during the init process. 
+
+# The Amplify cli artifacts
```

Comment:
Don't we also have a GraphQL file(s) that are created as metadata in addition to the schema?